src/main/java/org/xstefank/whitelist/WhitelistProcessing.java

```diff
@@ -0,0 +1,111 @@
+package org.xstefank.whitelist;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.xstefank.ci.CILoader;
+import org.xstefank.ci.ContinuousIntegration;
+import org.xstefank.api.GitHubAPI;
+import org.xstefank.model.PersistentList;
+import org.xstefank.model.TyrProperties;
+import org.xstefank.model.Utils;
+import org.xstefank.model.yaml.FormatConfig;
+
+public class WhitelistProcessing {
+
+    public static final boolean IS_WHITELISTING_ENABLED =
+            TyrProperties.getBooleanProperty(Utils.WHITELIST_ENABLED);
+
+    private static final List<ContinuousIntegration> continuousIntegrations = new ArrayList<>();
+
+    private final PersistentList userList;
+    private final PersistentList adminList;
+
+    private final List<Command> commands;
+
+    public WhitelistProcessing(FormatConfig config) {
+        String dirName = System.getProperty(Utils.JBOSS_CONFIG_DIR);
+        userList = new PersistentList(dirName, Utils.USERLIST_FILE_NAME);
+        adminList = new PersistentList(dirName, Utils.ADMINLIST_FILE_NAME);
+        commands = getCommands(config);
+        loadCIs(config);
+    }
+
+    public static String getCommentAuthor(JsonNode payload) {
+        return payload.get(Utils.COMMENT).get(Utils.USER).get(Utils.LOGIN).asText();
+    }
+
+    public static String getPRAuthor(JsonNode payload) {
+        return payload.get(Utils.ISSUE).get(Utils.USER).get(Utils.LOGIN).asText();
+    }
+
+    static void triggerCI(JsonNode payload) {
+        JsonNode prJson = GitHubAPI.getJsonWithPullRequest(payload);
+        for (ContinuousIntegration CI : continuousIntegrations) {
+            CI.triggerBuild(prJson);
+        }
+    }
+
+    static void loadCIs(FormatConfig config) {
+        List<String> CIConfigList = config.getFormat().getCI();
+
+        if (CIConfigList == null || CIConfigList.isEmpty()) {
+            return;
+        }
+
+        for (String key : CIConfigList) {
+            ContinuousIntegration CI = CILoader.getCI(key);
+            if (CI != null) {
+                CI.init();
+                continuousIntegrations.add(CI);
+            }
+        }
+    }
+
+    public void processPRComment(JsonNode payload) {
+        if (payload.get(Utils.ISSUE).has(Utils.PULL_REQUEST) &&
+                payload.get(Utils.ACTION).asText().matches("created") &&
+                !commands.isEmpty()) {
+            String message = payload.get(Utils.COMMENT).get(Utils.BODY).asText();
+            for (Command command : commands) {
+                if (message.matches(command.getCommandRegex())) {
+                    command.process(payload, adminList, userList);
+                }
+            }
+        }
+    }
+
+    public void triggerCIOnOpened(JsonNode payload) {
```

Comment:
it should be clear here that payload is different than the one that is being passed to triggerCI - I don't really see the need for two methods